scripts/build_funcs.sh

```diff
@@ -174,83 +190,78 @@ native() {
 gen_manifest_with_tag() {
-    imgTag="${2:-$GIT_REF_SHORT}"
-	docker manifest create \
-    "$REGISTRY/$TARGET:$1" \
-    --amend "$REGISTRY/$TARGET-amd64:$imgTag" \
-    --amend "$REGISTRY/$TARGET-arm64:$imgTag" 
+    imgTag="${2:-$IMAGE_VERSION}"
+
+    for arch in "${LINUX_ARCH[@]}"; do
+        docker manifest create \
+            "$REGISTRY/$TARGET:$1" \
+            --amend "$REGISTRY/$TARGET:$arch-$imgTag"
+    done
+    for VERSION in "${WINVERSIONS[@]}"; do
+        full_version=$(docker manifest inspect ${WIN_AMD64_BASEIMAGE}:${VERSION} | jq '.manifests[0].platform."os.version"' -r)
+        docker manifest create \
+            "$REGISTRY/$TARGET:$1" \
+            --amend "$REGISTRY/$TARGET:win-$WIN_ARCH-$VERSION-$imgTag"
+        docker manifest annotate "$REGISTRY/$TARGET:$1" \
+            "$REGISTRY/$TARGET:win-$WIN_ARCH-$VERSION-$imgTag" \
+            --os-version="${full_version}"
+    done
 }
 
 # Pushes the multi-os manifest for sonobuoy; must be generated first.
 push_manifest_with_tag() {
     gen_manifest_with_tag "$1"
-	docker manifest push "$REGISTRY/$TARGET:$1"
+    docker manifest push "$REGISTRY/$TARGET:$1"
 }
 
 # Pushes all images and the manifest.
 # Assumes you have the images built or loaded already. Not
 # added as dependency due to having both Linux/Windows
 # prereqs which can't be done on the same machine.
 gen_manifest_and_push_all() {
-	# Pushes the images first
-    if [ "$PUSH_WINDOWS" ] ; then
-        for arch in "${WIN_ARCH[@]}"; do
-            push_images TARGET="sonobuoy-win-$arch"
-        done
-    else
-        echo 'PUSH_WINDOWS not set, not pushing Windows images'
-        for arch in "${LINUX_ARCH[@]}"; do
-            push_images TARGET="sonobuoy-$arch"
-        done
-    fi
+    push_images
 
-	if git describe --tags --exact-match >/dev/null 2>&1 ; then
-		push_manifest_with_tag "$IMAGE_VERSION"
-		push_manifest_with_tag latest
-		push_manifest_with_tag "$IMAGE_TAG"
-        push_manifest_with_tag "$GIT_REF_SHORT"
+    if git describe --tags --exact-match >/dev/null 2>&1 ; then
+        push_manifest_with_tag "$IMAGE_VERSION"
+        push_manifest_with_tag "$IMAGE_BRANCH"
+        push_manifest_with_tag "$IMAGE_TAG"
+        push_manifest_with_tag latest
     else
-        push_manifest_with_tag "$GIT_REF_SHORT"
+        push_manifest_with_tag "$IMAGE_VERSION"
         push_manifest_with_tag "$IMAGE_BRANCH"
 	fi
 }
 
 # Removes a given image from docker. Image name (not registry) should be the first
 # and only argument.
 remove-image() {
-	docker rmi -f "$(docker images "$REGISTRY/$1" -a -q)" || true
+    docker rmi -f "$(docker images "$REGISTRY/$1" -a -q)" || true
 }
 
 # Removes temp files, built images, etc so the next build and repo are
 # in a pristine state.
 clean() {
     # Best effort for clean; don't exit if failure.
     set +e
-	rm -f "$TARGET"
-	rm Dockerfile-*
-	rm DockerfileWindows-*
-	rm -rf build
-
-	for arch in "${LINUX_ARCH[@]}"; do
```

Comment:
These should probably still be in place; the images are persisted as tarballs but they could be loaded locally and cause issues I presume. Going to add them back.

Reply:
Dur. Nm. Removing the image $TARGET is removing them all now. Thats why I removed them.